### PR TITLE
Updates to flags, b-tagging, backgrounds and signal, and datasets

### PR DESCRIPTION
**Pull request by gmadigan**
Made the number of b-tags configurable. Double check that the implementation of 0 and 2 btags (selection and SFs) looks right to you. Also, the flags are now consistent across years, and are all boolean, so I've cleaned up those assignments in the ntuple analyzer. Otherwise, I included the ttV and signal samples in the plotter, permanently added the linear GoodVertex plot, and updated the the ntuple info for 2016 and 2017.

### CODE REVIEW

File: LQResultProducer.py [hunk 1]
```diff
@@ -109,19 +111,36 @@
...
+deepJetWPmedium = '1'
+if year == '2016': deepJetWPmedium = '0.3093'
+elif year == '2017': deepJetWPmedium = '0.3033'
+elif year == '2018': deepJetWPmedium = '0.2770'
+
+if btags == '0':
+	bTagSFmedium = '*1'
+	bTagSFmediumUp = '*1'
+	bTagSFmediumDown = '*1'
+	bTagselmedium = '*1'
```
Comment (DavidMorse) [inline LQResultProducer.py]: The 0 btag case will need a b veto, i.e.

bTagselmedium = '*(((DeepJet_jet1>'+deepJetWPmedium+')+(DeepJet_jet2>'+deepJetWPmedium+'))==0)'

We will need to think about whether this requires an event weight, have to check the twikis

File: LQResultProducer.py [hunk 2]
```diff
@@ -109,19 +111,36 @@
...
+elif btags == '1':
+	bTagSFmedium = '*(1-(1-(DeepJet_jet1>'+deepJetWPmedium+')*bTagSF_jet1)*(1-(DeepJet_jet2>'+deepJetWPmedium+')*bTagSF_jet2))'
+	bTagSFmediumUp = '*(1-(1-(DeepJet_jet1>'+deepJetWPmedium+')*bTagSF_jet1Up)*(1-(DeepJet_jet2>'+deepJetWPmedium+')*bTagSF_jet2Up))'
+	bTagSFmediumDown = '*(1-(1-(DeepJet_jet1>'+deepJetWPmedium+')*bTagSF_jet1Down)*(1-(DeepJet_jet2>'+deepJetWPmedium+')*bTagSF_jet2Down))'
+	bTagselmedium = '*(((DeepJet_jet1>'+deepJetWPmedium+')+(DeepJet_jet2>'+deepJetWPmedium+'))>0)'
+elif btags == '2':
+	bTagSFmedium = '*(DeepJet_jet1>'+deepJetWPmedium+')*(DeepJet_jet2>'+deepJetWPmedium+')*(1-(1-bTagSF_jet1)*(1-bTagSF_jet2))'
+	bTagSFmediumUp = '*(DeepJet_jet1>'+deepJetWPmedium+')*(DeepJet_jet2>'+deepJetWPmedium+')*(1-(1-bTagSF_jet1Up)*(1-bTagSF_jet2Up))'
+	bTagSFmediumDown = '*(DeepJet_jet1>'+deepJetWPmedium+')*(DeepJet_jet2>'+deepJetWPmedium+')*(1-(1-bTagSF_jet1Down)*(1-bTagSF_jet2Down))'
+	bTagselmedium = '*(DeepJet_jet1>'+deepJetWPmedium+')*(DeepJet_jet2>'+deepJetWPmedium+')'
```
Comment (DavidMorse) [inline LQResultProducer.py]: Looks like this is missing '))>1)' instead of ')'

**Review by DavidMorse — CHANGES_REQUESTED** [review]
As a general comment, please note that the btag event weights built here are inclusive, i.e. >=1 or >=2. If we move to exclusive categories we will have to look again at how to correctly build the event weights.

File: LQResultProducer.py (hunk 1 above)
Comment (gmadigan) [reply LQResultProducer.py]: I see. So it would be exactly 0 b-tags, at least 1 b-tag, or exactly 2 b-tags (since we only consider the 2 leading jets).

Comment (DavidMorse) [reply LQResultProducer.py]: For now, yes. Eventually it should be ==0, ==1, ==2

**Pull request merged**